plugins/node/opentelemetry-instrumentation-mongodb/src/instrumentation.ts

```diff
@@ -527,15 +532,23 @@ export class MongoDBInstrumentation extends InstrumentationBase {
             cmd,
             commandType
           );
-          const patchedCallback = instrumentation._patchEnd(
+          patchedCallback = instrumentation._patchEnd(
             span,
             resultHandler,
             this.id,
             commandType
           );
-
-          return original.call(this, ns, cmd, options, patchedCallback);
         }
+
+        const result = original.call(this, ns, cmd, options, patchedCallback);
```

Comment:
Similar Q to above: would separate patch functions (one that uses the callback, one that uses the promise) be clearer? or would they just be a pain?

Reply:
here the `patchedCallback` can be reused because basically the API change has been to "promisify" the function. A separate patch is not complicated, it's almost an identical copy.